tutorials/data-science-preprocessing/clean.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Cleans Meteorite Landings json data.
+
+See http://catalog.data.gov/dataset/meteorite-landings-api to download the
+data this script is expecting.
+"""
+
+import argparse
+import json
+import re
+import textwrap
+
+import apache_beam
+from apache_beam.io import filebasedsource
+
+
+BIGQUERY_TABLE_FORMAT = re.compile(r'[\w.:-]+:\w+\.\w+$')
+
+class JsonFileSource(filebasedsource.FileBasedSource):
```

Comment:
This is good, and should probably be contributed back to Beam IMO!

Thoughts @robertwb ?